File: caesar-cipher.js
function caesarCipher(s, k) {
    let string = s
    console.log(string)
    let shift = k % 26
    const alphabet = Array.from({ length: 26 }, (_, i) => String.fromCharCode('a'.charCodeAt(0) + i));
    
    let result = ''
    
    for (let char of s) {
        if (alphabet.includes(char)) {
            const index = alphabet.indexOf(char)
            const newIndex = (index + shift) % 26
            result += alphabet[newIndex]
        } else if (alphabet.includes(char.toLowerCase())) {
            const index = alphabet.indexOf(char.toLowerCase())
            const newIndex = (index + shift) % 26
            result += alphabet[newIndex].toUpperCase()
        } else {
            result += char
        }
    }
    
    console.log(result)
    return result

}

File: caesar-cipher.py
import string

def caesarCipher(s, k):
    message = s
    shift = k % 26
    alphabet = list(string.ascii_lowercase)
    print(alphabet)
    code = ''
    
    for character in message:
        if character in alphabet:
            index = alphabet.index(character)
            newIndex = (index + shift) % 26
            code += alphabet[newIndex]
        elif character.lower() in alphabet:
            index = alphabet.index(character.lower())
            newIndex = (index + shift) % 26
            code += alphabet[newIndex].upper()
        else:
            code += character
    
    return code
